lib/optimizely.rb

```diff
@@ -258,11 +258,13 @@ def track(event_key, user_id, attributes = nil, event_tags = nil)
       @event_processor.process(user_event)
       @logger.log(Logger::INFO, "Tracking event '#{event_key}' for user '#{user_id}'.")
 
-      log_event = EventFactory.create_log_event(user_event, @logger)
-      @notification_center.send_notifications(
-        NotificationCenter::NOTIFICATION_TYPES[:TRACK],
-        event_key, user_id, attributes, event_tags, log_event
-      )
+      if @notification_center.notification_count(NotificationCenter::NOTIFICATION_TYPES[:TRACK]).positive?
```

Comment:
What do we gain by checking notification count, as opposed to just sending notifications regardless of count?

Reply:
This validates that user has already added the listener before sending the notification.
Please see java-sdk as a reference: https://github.com/optimizely/java-sdk/blob/master/core-api/src/main/java/com/optimizely/ab/Optimizely.java#L304